util/oteltest/src/oteltest/private.py

```diff
@@ -0,0 +1,283 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import glob
+import importlib
+import inspect
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+import time
+import typing
+import venv
+from pathlib import Path
+
+from google.protobuf.json_format import MessageToDict
+from oteltest import OtelTest, telemetry
+from oteltest.sink import GrpcSink, RequestHandler
+
+from opentelemetry.proto.collector.logs.v1.logs_service_pb2 import (
+    ExportLogsServiceRequest,
+)
+from opentelemetry.proto.collector.metrics.v1.metrics_service_pb2 import (
+    ExportMetricsServiceRequest,
+)
+from opentelemetry.proto.collector.trace.v1.trace_service_pb2 import (
+    ExportTraceServiceRequest,
+)
+
+
+def run(script_path: str, wheel_file: str, venv_parent_dir: str):
+    temp_dir = venv_parent_dir or tempfile.mkdtemp()
+    print(f"- Using temp dir for venvs: {temp_dir}")
+
+    if os.path.isdir(script_path):
+        handle_dir(script_path, temp_dir, wheel_file)
+    elif os.path.isfile(script_path):
+        handle_file(script_path, temp_dir, wheel_file)
+    else:
+        print(f"- {script_path} does not exist")
+        return
+
+
+def handle_dir(script_path, temp_dir, wheel_file):
+    sys.path.append(script_path)
+    for script in ls_scripts(script_path):
+        print(f"- Setting up environment for dir {script}")
+        setup_script_environment(temp_dir, script_path, script, wheel_file)
+
+
+def handle_file(script_path, temp_dir, wheel_file):
+    print(f"- Setting up environment for file {script_path}")
+    script_dir = os.path.dirname(script_path)
+    sys.path.append(script_dir)
+    setup_script_environment(
+        temp_dir, script_dir, os.path.basename(script_path), wheel_file
+    )
+
+
+def ls_scripts(script_dir):
+    original_dir = os.getcwd()
+    os.chdir(script_dir)
+    scripts = glob.glob("*.py")
+    os.chdir(original_dir)
+    return scripts
+
+
+def setup_script_environment(venv_parent, script_dir, script, wheel_file):
+    handler = AccumulatingHandler()
+    sink = GrpcSink(handler)
+    sink.start()
+
+    module_name = script[:-3]
+    oteltest_instance: OtelTest = load_test_class_for_script(module_name)()
```

Comment:
Something around here needs to be a bit more smart handling empty files:

```
- Setting up environment for __init__.py
Traceback (most recent call last):
  File "/home/rm/src/test-otel-python/venvtest/bin/oteltest", line 8, in <module>
    sys.exit(main())
  File "/home/rm/src/test-otel-python/venvtest/lib/python3.10/site-packages/oteltest/main.py", line 42, in main
    run(args.script_dir, args.wheel_file, args.venv_parent_dir)
  File "/home/rm/src/test-otel-python/venvtest/lib/python3.10/site-packages/oteltest/private.py", line 37, in run
    setup_script_environment(temp_dir, script_dir, script, wheel_file)
  File "/home/rm/src/test-otel-python/venvtest/lib/python3.10/site-packages/oteltest/private.py", line 54, in setup_script_environment
    oteltest_instance: OtelTest = load_test_class_for_script(module_name)()
TypeError: 'NoneType' object is not callable
```

Reply:
Thanks, fixed.